src/DqtApi/V2/Handlers/SetIttOutcomeHandler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DqtApi.DataStore.Crm;
+using DqtApi.V2.ApiModels;
+using DqtApi.V2.Requests;
+using DqtApi.V2.Responses;
+using DqtApi.Validation;
+using MediatR;
+
+namespace DqtApi.V2.Handlers
+{
+    public class SetIttOutcomeHandler : IRequestHandler<SetIttOutcomeRequest, SetIttOutcomeResponse>
+    {
+        private readonly IDataverseAdapter _dataverseAdapter;
+
+        public SetIttOutcomeHandler(IDataverseAdapter dataverseAdapter)
+        {
+            _dataverseAdapter = dataverseAdapter;
+        }
+
+        public async Task<SetIttOutcomeResponse> Handle(SetIttOutcomeRequest request, CancellationToken cancellationToken)
+        {
+            var teachers = (await _dataverseAdapter.GetTeachersByTrn(request.Trn, activeOnly: true)).ToArray();
+
+            if (teachers.Length == 0)
+            {
+                throw new ErrorException(10001);  // Teacher with specified TRN not found
```

Comment:
It feels like the error numbers more like 

```csharp
ErrorCode.TeacherWithSpecifiedTRNNotFound
```

Reply:
@defong I like that. I'll change them up.